### PR TITLE
Fix ipp tests

### PR DESCRIPTION
**Pull request by annawoodard**
This is partial (there are still other issues to be addressed, but I wanted to go ahead and merge what I've got so far).

### CODE REVIEW

File: parsl/tests/test_ipp/bash_app.py [hunk 1]
```diff
@@ -12,12 +12,15 @@
 # parsl.set_stream_logger()
 
 workers = IPyParallelExecutor()
-dfk = DataFlowKernel(workers)
+dfk = DataFlowKernel(executors=[workers])
```
Comment (yadudoc) [inline parsl/tests/test_ipp/bash_app.py]: Sorry these tests have not been updated!

File: parsl/tests/test_ipp/bash_app.py [hunk 2]
```diff
@@ -12,12 +12,15 @@
...
 
 
 @App('bash', dfk)
 def echo_to_file(inputs=[], outputs=[], stderr='std.err', stdout='std.out'):
-    pass
+    res = ""
+    for i, o in zip(inputs, outputs):
```
Comment (yadudoc) [inline parsl/tests/test_ipp/bash_app.py]: This assumesj len(inputs) == len(outputs).
Might be safer to do : 
return "for i in {0}; do echo {1} &> $i ; done".format(" ".join(outputs), " ".join(inputs) )

File: parsl/tests/test_ipp/bash_app.py [hunk 3]
```diff
@@ -64,12 +67,12 @@ def test_parallel_for(n=10):
 
     start = time.time()
     for i in range(0, n):
-        d[i], _ = echo_to_file(inputs=['Hello World {0}'.format(i)],
```
Comment (yadudoc) [inline parsl/tests/test_ipp/bash_app.py]: sorry! ancient code.

File: parsl/tests/test_ipp/bash_app.py (hunk 2 above)
Comment (annawoodard) [reply parsl/tests/test_ipp/bash_app.py]: Good point @yadudoc, I've updated it.

**Pull request merged**